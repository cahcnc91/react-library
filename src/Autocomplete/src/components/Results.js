import React from 'react';
import { createUseStyles } from "react-jss";
import ResultItem from './ResultItem';

const useStyles = createUseStyles({
  fullHistory: {
    cursor: 'pointer',
    padding: '.5rem 1.1rem',
    '&:hover': {
      backgroundColor: 'rgba(0, 0, 0, .06)'
    }
  },
  boxResultsWrapper: {
    position: 'absolute',
    zIndex: 10,
    width: '100%',
    padding: '.5rem 0',
    borderRadius: 5,
    borderTop: 0,
    borderTopLeftRadius: 0,
    borderTopRightRadius: 0,
    boxShadow: '0 2px 4px rgba(0,0,0,0.2)'
  },
  noResults: {
    padding: ".3rem 1.1rem",
    fontFamily: 'Helvetica, sans-serif'
  }
});

const Results = ({
  showResults,
  results,
  activeOption,
  changeActiveOption,
  handleClickItem
}) => {
  const classes = useStyles();

  return (
    <div className={showResults ? classes.boxResultsWrapper : ''}>
      {showResults && results.length > 0 &&
        results.map((item, index) => (
            <ResultItem
              key={`${item.id}`}
              item={item}
              handleClickItem={handleClickItem}
              changeActiveOption={changeActiveOption}
              index={index}
              activeOption={activeOption}
            />
        ))}
        {showResults && results.length === 0 &&
        <div className={classes.noResults}> No Results found </div>
        }
    </div>
  );
};

export default Results;
