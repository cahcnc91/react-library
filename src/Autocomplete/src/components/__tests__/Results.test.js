import React from "react";
import { render } from "enzyme";
import Results from "../Results";

const handleClickItem = () => {
};

const changeActiveOption = () => {
};

const languages = [
    {
      id: 1,
      label: "C",
      year: 1972
    },
    {
      id: 2,
      label: "C#",
      year: 2000
    },
    {
      id: 3,
      label: "C++",
      year: 1983
    },
    {
      id: 4,
      label: "Clojure",
      year: 2007
    }
  ];

describe("Results", () => {
  let component;
  beforeAll(() => {
    component = render(
      <Results
        showResults={true}
        results={languages}
        activeOption={1}
        handleClickItem={handleClickItem}
        changeActiveOption={changeActiveOption}
      />
    );
  });
  test("snapshot renders", () => {
    let tree = component.html();
    expect(tree).toMatchSnapshot();
  });

  it("renders wrapper and all list options", () => {
    // tests wrapper class
    expect(component.find('Results-boxResultsWrapper-0-2-2')).toBeDefined();
    expect(component.children()).toHaveLength(languages.length);
  });
});
