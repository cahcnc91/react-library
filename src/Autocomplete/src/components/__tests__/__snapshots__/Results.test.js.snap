// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results snapshot renders 1`] = `"<div class=\\"nonSelected-0-2-4\\" data-test=\\"result-item-1\\">C</div><div class=\\"selected-0-2-5\\" data-test=\\"result-item-2\\">C#</div><div class=\\"nonSelected-0-2-4\\" data-test=\\"result-item-3\\">C++</div><div class=\\"nonSelected-0-2-4\\" data-test=\\"result-item-4\\">Clojure</div>"`;
